api/v1alpha2/conversion.go

```diff
@@ -120,6 +122,8 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	v1alpha3.ExcludeNodeDrainingAnnotation = "machine.cluster.x-k8s.io.io/exclude-node-draining"
```

Comment:
You can't change the value of a constant, nor do we want to 😄. Instead, please add logic that does the following:

- If converting from v1alpha2 to v1alpha3, if there is an annotation in the v1alpha2 machine with the key `machine.cluster.x-k8s.io.io/exclude-node-draining`, please change it to `machine.cluster.x-k8s.io/exclude-node-draining` and delete the other key
- If converting from v1alpha3 to v1alpha2, if there is an annotation in the v1alpha3 machine with the key `machine.cluster.x-k8s.io/exclude-node-draining`, please change it to `machine.cluster.x-k8s.io.io/exclude-node-draining` and delete the other key

right @vincepri?

Reply:
Is there an issue with changing the constant in the v1alpha3 version of the types, and using the respective constants (v1alpha2 vs v1alpha3 types) to do the comparisons in the conversion?

Reply:
I see the constant has been changed in the v1alpha3 file, we only need to convert the annotation in `ConvertTo` and `ConvertFrom` methods.

Converting from v1alpha3:
- If the `v1alpha3. ExcludeNodeDrainingAnnotation` is present on the v1alpha3 object in the `Annotations` map, change it to `(v1alpha2.)ExcludeNodeDrainingAnnotation`.

Converting to v1alpha3:
- If the `(v1alpha2.)ExcludeNodeDrainingAnnotation` is present on the v1alpha2 object in the `Annotations` map, change it to `v1alpha3.ExcludeNodeDrainingAnnotation`.

Reply:
Updated. PTAL! Thanks